### ISSUE TITLE
Questions about Minecraft PE chunks

### ISSUE THREAD

**Issue by robotman3000** [DragonetMC/DragonProxy#40]
Hello, me and @MrPowerGamerBR have been trying for [some time now](https://github.com/robotman3000/DragonProxy/issues/5) to get the chunk data that DragonProxy sends to the minecraft pe client to be rendered. The client either crashes or appears to ignore the chunk data , so my question is: In the MCPE protocol, assuming the ChunkData packet has been encoded properly, what other things does the client look for before it loads chunks?

**Comment by DefinitlyEvil** [comment]
first 3 ppl you tagged won't possibly reply lol

**Comment by Creeperface01** [comment]
ChunkRadiusUpdatedPacket maybe :D

**Comment by Creeperface01** [comment]
@NiclasOlofsson

**Comment by inxomnyaa** [comment]
Well, good that i follow the repository. Chunks are divided in subchunks, but you could also send a full column (thats like the highlighted chunk in MCPC with f3+sth.., 0 to max)
First of all, the block translator is impossible.. though mcpe doesnt crash on unknown blocks anymore, its still a todo.
> ChunkRadiusUpdatedPacket maybe :D 

Yes, this one too. (And requestradiuspacket (sth like that)). Its the render distance and has nothing to do with "server pllzzzz send me the chunk"

If i would understand how to set up the new dragonproxy version, i would like to try to make it work, but currently all i have is confusion

**Comment by Creeperface01** [comment]
But If you dont send chunk radius packet it might (note by @thebigsmileXD: **will**) wont work

**Comment by NiclasOlofsson** [comment]
You need to deal with the resource packets too. There are multiple of them. Without it, the client doesn't spawn properly.

I would love to help. I do. But I don't like this kind of spoon-feeding. There are way to many servers out there working just fine. When you ask "questions" the way you do, you actually require the people you ask to get into YOUR code. I suggest you *at least try* to get into THEIR code first. Your questions should then be formulated like "I've tried this, and that ... This is what my code is doing now ... This is my problem ... I had a look at your code and I have a question about this specific thingy .... "

**Comment by DefinitlyEvil** [comment]
@NiclasOlofsson thanks man 
@thebigsmileXD so chunks are sent more like PC way? column mask?

**Comment by inxomnyaa** [comment]
@DefinitlyEvil a little bit.. @NiclasOlofsson yea sure, but you can simply reply with an empty resource pack array, then you only need 1 packet of those

**Comment by MrPowerGamerBR** [comment]
I thought this issue was about sending a chunk packet not random unrelated packet things. (getting a player to join a server is easy, successfully sending a chunk packet is difficult)

Also, I'm not the current developer of DragonProxy, after giving up trying to send a chunk packet I decided to fork Nukkit and use my own fork for my server and, due to college and stuff, I may not have time to work on DragonProxy again.

TL;DR: Chunks are retarded

**Comment by DaMatrix** [comment]
I'm not sure how this conversation moved to login, #38 fixed that (the player can join in a void world and recieve PC chat packets). The issue was with chunks :/

**Comment by robotman3000** [comment]
@NiclasOlofsson Sorry, I wasn't meaning to come across like that at all. Since there is a lack of good MCPE protocol documentation I thought that I would ask questions and compile good documentation as DragonProxy is repaired. Also I am looking for answers to my questions that are independent of any code base. So, let me rephrase the question to better reflect what I really meant. 

These questions are for anyone who can answer
In the MCPE protocol, assuming the ChunkData packet has been encoded properly, what other things does the client look for before it loads chunks? 

Here are some expansions of this question: 
Do I have to send the ChunkRadius packet?
@NiclasOlofsson Can you explain more about the resource packets and the kind of problems that could occur?
Does the StartGame packet's data have any effect on chunk loading?
Do packets have to be sent in a particular order?

Thanks

**Comment by robotman3000** [comment]
I have updated the initial comment with the new question

**Comment by robotman3000** [comment]
@thebigsmileXD Would you please create an issue and explain the problems you are having? Then  can try to help you solve your problems.

**Comment by robotman3000** [comment]
@DefinitlyEvil Are they not active on github, or just too busy?

**Comment by DefinitlyEvil** [comment]
@robotman3000 they just don't wanna talk to ppl

**Comment by ghost** [comment]
> when wil it work the chunks

When will the chunks work

**Comment by inxomnyaa** [comment]
@caspervanneck don't ask for ETA ;)

**Comment by inxomnyaa** [comment]
https://sel-project.github.io/sel-utils/pocket/101.html

**Comment by inxomnyaa** [comment]
So over all, all thats left to do is sending the chunks?

**Comment by DaMatrix** [comment]
Well, once we've got chunks working we can get around to fixing the rest of the translators. Right now all DragonProxy can do is connect and receive chat.

**Comment by inxomnyaa** [comment]
Seems like i do something wrong. I get stuck at an empty 'dirt' screen when trying to connect to my spigot server

**Comment by MrPowerGamerBR** [comment]
So easy

![´http://i.imgur.com/8UwxVVf.png](http://i.imgur.com/8UwxVVf.png)

**Comment by MrPowerGamerBR** [comment]
I must agree with Niclas now, I wasn't able to get those chunks working, so I stopped developing DragonProxy and started messing around with Nukkit's source code, learned a lot about MCPE packets and RakNet, helped fixing issues in Nukkit and JRakNet and, after a few days, I finally decided to try to send a chunk packet again.

And, guess what? It is difficult? Yeah, but if you try to read how the code works, it is easy.

![http://i.imgur.com/7c72KVX.png](http://i.imgur.com/7c72KVX.png)

The only time I had to ask something was when the client was receiving the chunk data however no blocks were rendered. (the fix was to surround the client with more chunk data packets)

**Comment by robotman3000** [comment]
@MrPowerGamerBR Looks great man. It took some time but the problem is solved now. Can you tell how you did it?

**Comment by robotman3000** [comment]
@MrPowerGamerBR Also how did you get the cordinates to display in the bottom right corner?

**Comment by DaMatrix** [comment]
There's plenty of different resource packs that can show your coordinates on-screen.

**Comment by MrPowerGamerBR** [comment]
@robotman3000 https://github.com/PocketDreams/SequinLand-Glowstone

There even two functions that I made to get an empty chunk and getting an empty subsection

**Comment by MrPowerGamerBR** [comment]
https://github.com/PocketDreams/SequinLand-Glowstone/blob/master/src/main/java/net/pocketdreams/sequinland/util/PocketChunkUtils.java

**Comment by MrPowerGamerBR** [comment]
Also I already got chunks that aren't just dirt chunks working...

![http://i.imgur.com/fVFWteq.png](http://i.imgur.com/fVFWteq.png)

**Comment by TheDiamondYT1** [comment]
Chunks are gay, put simply.

**Comment by inxomnyaa** [comment]
Chunks can't be gay. They don't live. Also, as "insult" or "comparasion" gay is the wrong term. Stay with "shitty"

**Comment by MrPowerGamerBR** [comment]
@thebigsmileXD they aren't shitty, they are just a tad confusing.

Also, you don't need to reply the client with the "UpdatedChunkRadiusPacket", the client doesn't care about that, that's only used to change the client's view distance.

Actually, you don't need to send anything except the chunk data packet for the client to be able to view the chunk.

**Comment by inxomnyaa** [comment]
@MrPowerGamerBR I was not saying they are shitty, i was correcting his terms of language.

**Comment by MrPowerGamerBR** [comment]
Also, a small shameless plug for my project (that may or may not help DragonProxy): https://github.com/PocketDreams/SequinLand-Glowstone

Yes, I'm trying to do what Dragonet tried to do a few years ago, the difference now? Well... I'm the one who is leading the project (and I have a bit of experience with PE packets) and, instead of just acting like an proxy (yeah, Dragonet **had** a DragonProxy inside of it to translate the packets!) I'm implementing PE support inside the code without messing around with translators.

I still think that implementing DragonProxy will be difficult, the PE protocol has some key differences that makes it difficult to implement by "translating" the packets.

**Comment by MrPowerGamerBR** [comment]
Also, a small shameless 
## **_http://lmgtfy.com/?q=plug_**
 for my project (that may or may not help DragonProxy): https://github.com/PocketDreams/SequinLand-Glowstone

Yes, I'm trying to do what Dragonet tried to do a few years ago, the difference now? Well... I'm the one who is leading the project (and I have a bit of experience with PE packets) and, instead of just acting like an proxy (yeah, Dragonet **had** a DragonProxy inside of it to translate the packets!) I'm implementing PE support inside the code without messing around with translators.

I still think that implementing DragonProxy will be difficult, the PE protocol has some key differences that makes it difficult to implement by "translating" the packets.

**Comment by DefinitlyEvil** [comment]
It needs to cache stuffs to be compatible with both protocols.

**Comment by DaMatrix** [comment]
...which almost makes it better to simply run the pocket protocol on the server itself.
